packages/cli/src/commands/org/config/preview.ts

```diff
@@ -0,0 +1,79 @@
+import {Command, flags} from '@oclif/command';
+import {cli} from 'cli-ux';
+import {ReadStream} from 'fs';
+import {cwd} from 'process';
+import {Config} from '../../../lib/config/config';
+import {
+  IsAuthenticated,
+  Preconditions,
+} from '../../../lib/decorators/preconditions';
+import {Project} from '../../../lib/project/project';
+import {SnapshotFactory} from '../../../lib/snapshot/snapshotFactory';
+
+export interface CustomFile extends ReadStream {
+  type?: string;
+}
+
+export default class Preview extends Command {
+  static description = 'Preview resource updates';
+
+  static flags = {
+    target: flags.string({
+      char: 't',
+      description:
+        'The unique identifier of the organization where to send the changes. If not specified, the organization you are connected to will be used.',
+      helpValue: 'destinationorganizationg7dg3gd',
+      required: false,
+    }),
+    projectPath: flags.string({
+      char: 'p',
+      description: 'The path to your Coveo project.',
+      helpValue: '/Users/Me/my-project',
+      default: cwd(),
+      required: false,
+    }),
+  };
+
+  static hidden = true;
+
+  @Preconditions(IsAuthenticated())
+  async run() {
+    const {flags} = this.parse(Preview);
+    const project = new Project(flags.projectPath);
+    const pathToZip = await project.compressResources();
+
+    const factory = new SnapshotFactory();
+    const snapshot = await factory.createFromZip(
+      pathToZip,
+      'cli-preview-snapshot'
+    );
+
+    cli.action.start('Creating snapshot');
+
+    const targetOrg = await this.getTargetOrg();
```

Comment:
Should there be a confirmation message here ?

`You are about to create and preview a snapshot for org "foo" in region "bar", do you wish to continue ?`

Or would that be too annoying ? Just to make sure that you are doing the operation in the correct org, like if you logged into an incorrect one...

Reply:
I think we should not, too much of a hassle.

I would reserve such warning for
 - Destructive operations
 - Operations having impact on the queries of the org (e.g. if an operation change a query pipeline in the org). So when you apply snapshot

Reply:
I agree with @louis-bompart Snapshot should not apply any modification to the destination org.
However, there will be a prompt if the snapshot requires a synchronization plan.